File: SolucionEscuelaBackend/Escuela.Domain/Entities/Student.cs
using System;
using System.Collections.Generic;

namespace Escuela.Domain.Entities;

public partial class Student
{
    public int Id { get; set; }

    public string FirstName { get; set; } = null!;

    public string LastName { get; set; } = null!;

    public DateOnly BirthDate { get; set; }

    public string Email { get; set; } = null!;

    public string Address { get; set; } = null!;

    public string Phone { get; set; } = null!;

    public string City { get; set; } = null!;

    public string Password { get; set; } = null!;

    public string UserName { get; set; } = null!;

    public DateTime CreationDate { get; set; }

    public bool Active { get; set; }

    public virtual ICollection<StudentXsubject> StudentXsubjects { get; set; } = new List<StudentXsubject>();
}

File: EscuelaApp/src/app/interfaces/student.ts
import { Subject } from "./subject";

export interface StudentRegister{
    firstName:string;
    lastName:string;
    documentNumber:string,
    city:string;
    email:string;
    address:string;
    phone:string;
    password:string;
}

export interface Student extends StudentRegister{
    id:number;
    userName:string;
    credits:number;
    active:boolean;
    creationDate:Date;
    subjects:Subject[];

}

